report/junit_test.go

```diff
@@ -0,0 +1,107 @@
+package report
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+)
+
+func TestWriteJunit(t *testing.T) {
+	tests := []struct {
+		findings       []Finding
+		testReportName string
+		expected       string
+		wantEmpty      bool
+	}{
+		{
+			testReportName: "simple",
+			expected:       filepath.Join(expectPath, "report", "junit_simple.xml"),
+			findings: []Finding{
+				{
+
+					Description: "Test Rule",
+					RuleID:      "test-rule",
+					Match:       "line containing secret",
+					Secret:      "a secret",
+					StartLine:   1,
+					EndLine:     2,
+					StartColumn: 1,
+					EndColumn:   2,
+					Message:     "opps",
+					File:        "auth.py",
+					Commit:      "0000000000000000",
+					Author:      "John Doe",
+					Email:       "johndoe@gmail.com",
+					Date:        "10-19-2003",
+					Tags:        []string{},
+				},
+				{
+
+					Description: "Test Rule",
+					RuleID:      "test-rule",
+					Match:       "line containing secret",
+					Secret:      "a secret",
+					StartLine:   2,
+					EndLine:     3,
+					StartColumn: 1,
+					EndColumn:   2,
+					Message:     "",
+					File:        "auth.py",
+					Commit:      "",
+					Author:      "",
+					Email:       "",
+					Date:        "",
+					Tags:        []string{},
+				},
+			},
+		},
+		{
+			testReportName: "empty",
+			expected:       filepath.Join(expectPath, "report", "junit_empty.xml"),
+			findings:       []Finding{},
+		},
+	}
+
+	for _, test := range tests {
+		// create tmp file using os.TempDir()
+		tmpfile, err := os.Create(filepath.Join(tmpPath, test.testReportName+".xml"))
```

Comment:
@zricethezav `report_test.go` declares `tmpPath` but this file doesn't (re: comment on Discord).

https://github.com/maltemorgenstern/gitleaks/blob/fade1e9ad47e66a233bc4ea1af6d3b46530e4f94/report/report_test.go#L12-L15

Reply:
can confirm tmpPath is not declared.

Reply:
fix't in https://github.com/gitleaks/gitleaks/commit/b1d59bf8905b3dae1b1f84afa76b0e7c6293d59e